src/net/JNetReflector/JNetReflectedCore.cs

```diff
@@ -0,0 +1,177 @@
+﻿/*
+*  Copyright 2022 MASES s.r.l.
+*
+*  Licensed under the Apache License, Version 2.0 (the "License");
+*  you may not use this file except in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*  http://www.apache.org/licenses/LICENSE-2.0
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+*
+*  Refer to LICENSE for more information.
+*/
+
+using MASES.CLIParser;
+using MASES.JNet;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace MASES.JNetReflector
+{
+    /// <summary>
+    /// Public entry point of <see cref="JNetReflectedCore{T}"/>
+    /// </summary>
+    public abstract class JNetReflectedCore<T> : JNetCoreBase<T> where T : JNetReflectedCore<T>
+    {
+        #region Initialization
+        /// <inheritdoc cref="JNetCoreBase{T}.CommandLineArguments"/>
+        public override IEnumerable<IArgumentMetadata> CommandLineArguments
+        {
+            get
+            {
+                var lst = new List<IArgumentMetadata>(base.CommandLineArguments);
+                lst.AddRange(new IArgumentMetadata[]
+                {
+                    new ArgumentMetadata<string>()
+                    {
+                        Name = CLIParam.OriginRootPath,
+                        Type = ArgumentType.Double,
+                        IsMandatory = true,
+                        Help = "Set the origin path where Jars are stored",
+                    },
+                    new ArgumentMetadata<string>()
+                    {
+                        Name = CLIParam.OriginJavadocUrl,
+                        Type = ArgumentType.Double,
+                        Default = null,
+                        Help = "The URL of the Javadoc to be associated to the classes",
+                    },
+                    new ArgumentMetadata<string>()
+                    {
+                        Name = CLIParam.DestinationRootPath,
+                        Type = ArgumentType.Double,
+                        Default = SpecialNames.JNetReflectorGeneratedFolder,
+                        Help = "Set the destination root path",
+                    },
+                    new ArgumentMetadata<string>()
+                    {
+                        Name = CLIParam.JarList,
+                        Type = ArgumentType.Double,
+                        Help = "A CSV list, separated with comma, of Jar to be analyzed",
```

Comment:
From the code seems it is possible to use both folders containing Jars or list of explicit Jars.

Reply:
Yes, the text shall be more explicit on this